### PR TITLE
new video elements

### PR DESCRIPTION
**Pull request by collin**
@kand @daytonn Left some notes for review.

### CODE REVIEW

File: elements/bulbs-video/bulbs-video.js [hunk 1]
```diff
@@ -16,23 +14,57 @@ import 'videojs/dist/video-js/video-js.css';
...
-    if (this.props.src !== props.src) {
+  componentDidUpdate (prevProps) {
+    if (this.props.src !== prevProps.src) {
       this.store.actions.resetController();
-      this.store.actions.fetchVideo(props.src);
+      this.initialDispatch();
+    }
+  }
+
+/*
```
Comment (collin) [inline elements/bulbs-video/bulbs-video.js]: Next task for me related to the video player involves getting redundant requests for the same resource trimmed down. Going to send out an RFC before diving into that in earnest.

File: elements/bulbs-video/bulbs-video.js [hunk 2]
```diff
@@ -47,13 +79,15 @@ Object.assign(BulbsVideo, {
...
+    noEndcard: PropTypes.string,
     src: PropTypes.string.isRequired,
     twitterHandle: PropTypes.isRequired,
   },
 });
 
 registerReactElement('bulbs-video', BulbsVideo);
+
+import './elements/meta';
+import './elements/summary';
```
Comment (collin) [inline elements/bulbs-video/bulbs-video.js]: I'm not sure if it makes sense to move `<bulbs-video-meta>` and `<bulbs-video-summary>` to their own top-level.

Thoughts?

File: elements/bulbs-video/components/cover.test.js [hunk 3]
```diff
@@ -51,7 +51,7 @@ describe('<bulbs-video> <Cover>', function () {
       );
     });
 
-    it('renders a CroppedImage', () => {
+    it('renders an image', () => {
```
Comment (collin) [inline elements/bulbs-video/components/cover.test.js]: As you may know there are problems with using CroppedImage in bulbs-elements.

The local image.js takes over and starts rendering the wrong images.

File: elements/bulbs-video/elements/meta.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
+import React, { PropTypes } from 'react';
```
Comment (collin) [inline elements/bulbs-video/elements/meta.js]: @daytonn From a few weeks later I'm not sure how vital it is to separate the outermost class from the 'Root' class. Probably not.

File: elements/bulbs-video/bulbs-video.js (hunk 2 above)
Comment (daytonn) [inline elements/bulbs-video/bulbs-video.js]: We can keep it here until it's obvious.

Comment (kand) [inline elements/bulbs-video/bulbs-video.js]: What's the distinction here between elements and components? Is one a function and the other a class?

File: elements/bulbs-video/components/cover.test.js (hunk 3 above)
Comment (kand) [inline elements/bulbs-video/components/cover.test.js]: @daytonn weren't we able to use CroppedImage?

File: elements/bulbs-video/elements/meta.js (hunk 4 above)
Comment (daytonn) [inline elements/bulbs-video/elements/meta.js]: Not sure what you mean.

File: elements/bulbs-video/components/revealed.js [hunk 5]
```diff
@@ -93,6 +93,10 @@ export default class Revealed extends React.Component {
       twitterHandle: this.props.twitterHandle,
     };
 
+    if (this.props.noEndcard) {
+      delete playerOptions.pluginConfig.endcard;
```
Comment (kand) [inline elements/bulbs-video/components/revealed.js]: This feels like an anti-pattern. If we don't want an endcard, couldn't we not pass in the endcard config?

File: elements/bulbs-video/bulbs-video.js (hunk 2 above)
Comment (collin) [inline elements/bulbs-video/bulbs-video.js]: Elements:

* registered as dom elements with `document.registerElement`
* have a special wrapper that marries `document.registerElement` with `ReactDOM.render`

Components
* mere react components (both classes and functions)

File: elements/bulbs-video/elements/meta.js (hunk 4 above)
Comment (collin) [inline elements/bulbs-video/elements/meta.js]: I was pushing a hard line on  creating a `Root` component for elements that separates store wiring stuff from top-level rendering logic.

Now that I'm going through and implementing more elements I'm not so attached to that and can't make a cogent argument for it.

File: elements/bulbs-video/components/revealed.js (hunk 5 above)
Comment (collin) [inline elements/bulbs-video/components/revealed.js]: These are options coming down through from the videohub JSON api.

**Comment by kand** [comment]
👍

**Comment by collin** [comment]
👍

**Comment by daytonn** [comment]
👍

**Pull request merged**